packages/core-mobile/app/new/routes/(signedIn)/(modals)/accountSettings/securityAndPrivacy.tsx

```diff
@@ -34,43 +31,28 @@ const SecurityAndPrivacyScreen = (): JSX.Element => {
   const coreAnalyticsConsent = useSelector(selectCoreAnalyticsConsent)
   const { allApprovedDapps } = useConnectedDapps()
   const walletType = useSelector(selectWalletType)
-  const [biometricType, setBiometricType] = useState<BiometricType>(
-    BiometricType.NONE
-  )
+  const activeWalletId = useSelector(selectActiveWalletId)
+  const {
+    biometricType,
+    isBiometricAvailable,
+    useBiometrics,
+    setUseBiometrics
+  } = useStoredBiometrics()
   const { navigate } = useRouter()
 
-  const [isBiometricSwitchEnabled, setIsBiometricSwitchEnabled] =
-    useState(false)
-  const [isBiometricAvailable, setIsBiometricAvailable] = useState(false)
-
-  useFocusEffect(
-    useCallback(() => {
-      BiometricsSDK.canUseBiometry()
-        .then((biometricAvailable: boolean) => {
-          setIsBiometricAvailable(biometricAvailable)
-        })
-        .catch(Logger.error)
-
-      const type = commonStorage.getString(StorageKey.SECURE_ACCESS_SET)
-      if (type) {
-        setIsBiometricSwitchEnabled(type === 'BIO')
-      } else {
-        Logger.error('Secure access type not found')
-      }
-    }, [])
-  )
-
   const handleSwitchBiometric = useCallback(
     (value: boolean): void => {
-      setIsBiometricSwitchEnabled(value)
+      setUseBiometrics(value)
       if (value) {
         // @ts-ignore TODO: make routes typesafe
         navigate('/accountSettings/biometricVerifyPin')
       } else {
-        commonStorage.set(StorageKey.SECURE_ACCESS_SET, 'PIN')
+        if (activeWalletId) {
```

Comment:
should we create a hook `useActiveWallet` or `useActiveWalletId` that will throw an error when there is no active wallet? that way the hook can always return a valid wallet/walletId and in the component we don't have to do these if checks.